accordion/accordion.py

```diff
@@ -1,107 +1,92 @@
-"""TO-DO: Write a description of what this XBlock is."""
+"""
+An XBlock for creating and accordion component with multiple panels with rich content.
+"""
 
-import pkg_resources
+import importlib_resources
 from django.utils import translation
 from web_fragments.fragment import Fragment
 from xblock.core import XBlock
-from xblock.fields import Integer, Scope
-from xblockutils.resources import ResourceLoader
+from xblock.fields import Dict, List, Scope, String
 
 
 class AccordionXBlock(XBlock):
     """
-    TO-DO: document what your XBlock does.
+    Accordion XBlock.
     """
 
-    # Fields are defined on the class.  You can access them in your code as
-    # self.<fieldname>.
-
-    # TO-DO: delete count, and define your own fields.
-    count = Integer(
-        default=0, scope=Scope.user_state,
-        help="A simple counter, to show something happening",
+    display_name = String(default=translation.gettext_noop("Accordion"))
+    panels = List(help="Accordion entries", default=[], scope=Scope.content)
+    styling = Dict(help="Accordion styling", default=[], scope=Scope.content)
+    border_style = String(
+        help="Accordion border style", default="", scope=Scope.content
     )
 
     def resource_string(self, path):
         """Handy helper for getting resources from our kit."""
-        data = pkg_resources.resource_string(__name__, path)
-        return data.decode("utf8")
+        data = importlib_resources.files("accordion").joinpath(path).read_text("utf8")
+        return data
 
-    # TO-DO: change this view to display your data your own way.
-    def student_view(self, context=None):
+    def student_view(self, context=None):  # pylint: disable=unused-argument
         """
         Create primary view of the AccordionXBlock, shown to students when viewing courses.
         """
-        if context:
-            pass  # TO-DO: do something based on the context.
-        html = self.resource_string("static/html/accordion.html")
-        frag = Fragment(html.format(self=self))
-        frag.add_css(self.resource_string("static/css/accordion.css"))
-
-        # Add i18n js
-        statici18n_js_url = self._get_statici18n_js_url()
-        if statici18n_js_url:
-            frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
-
-        frag.add_javascript(self.resource_string("static/js/src/accordion.js"))
-        frag.initialize_js('AccordionXBlock')
+        frag = Fragment()
+        frag.add_javascript(self.resource_string("static/student.js"))
+        frag.add_css_url(self.runtime.local_resource_url(self, "public/student-ui.css"))
+        frag.initialize_js(
+            "AccordionBlock",
+            {
+                "url": self.runtime.local_resource_url(self, "public/student-ui.js"),
```

Comment:
We pass the URL of the compiled module to a minimal function that will simply import it to initialize it.